app/services/lti_membership_service.rb

```diff
@@ -0,0 +1,66 @@
+class LtiMembershipService
+  def initialize(lti_configuration)
+    raise "given LtiConfiguration must exist" unless lti_configuration.present?
+
+    @lti_configuration = lti_configuration
+  end
+
+  # TODO: this method is probably entirely unnecessary
+  def get_membership_from_nonce(nonce)
+    message_store = GitHubClassroom.lti_message_store(
+      consumer_key: lti_configuration.consumer_key
+    )
+    launch_message = message_store.get_message(nonce)
+    auth_params = launch_message.oauth_params
+    custom_params = launch_message.custom_params
+
+    membership_service_url = custom_params[:custom_context_memberships_url]
+    return nil unless membership_service_url
+
+    get_membership(membership_service_url)
+  end
+
+  def get_membership(endpoint)
+    response = make_signed_membership_request(endpoint)
+
+    raise 'error fetching membership' unless response.status == 200
+    membership_container = JSON.parse(response.body)
+
+    build_memberships_list(membership_container)
+  end
+
+  def build_memberships_list(membership_container)
+    memberships_list = []
+
+    memberships = membership_container.dig("pageOf", "membershipSubject", "membership")
+    memberships.each do |membership|
+      symbolized_membership = membership.deep_transform_keys { |key| key.underscore.to_sym }
+      symbolized_member = symbolized_membership[:member]
+
+      parsed_member = IMS::LTI::Models::MembershipService::LISPerson.new(symbolized_member)
+      symbolized_membership[:member] = parsed_member
+
+      parsed_membership = IMS::LTI::Models::MembershipService::Membership.new(symbolized_membership)
+      memberships_list.push(parsed_membership)
+    end
+
+    memberships_list
+  end
+
+  def make_signed_membership_request(endpoint)
```

Comment:
I'm not a fan of having to manually sign any outgoing requests, but the LTI standard requires a little-used extension of OAuth 1.0 called the `oauth_body_hash` to be present when generating the signature, and I don't know of any mainstream OAuth libraries makes it as simple as a boolean -- so I used ruby's [go-to OAuth library](https://rubygems.org/gems/oauth/versions/0.5.1) to generate it.

This method should probably be abstracted to a generic `LtiRequestService` class so this method can be reused for future lti services, too.